go/enclave/enclave_admin_service.go

```diff
@@ -187,10 +187,13 @@ func (e *enclaveAdminService) SubmitL1Block(ctx context.Context, blockData *comm
 		return nil, e.rejectBlockErr(ctx, fmt.Errorf("could not submit L1 block. Cause: %w", err))
 	}
 
-	bsr := &common.BlockSubmissionResponse{
-		ProducedSecretResponses: e.sharedSecretProcessor.ProcessNetworkSecretMsgs(ctx, blockData),
-		RollupMetadata:          rollupMetadata,
+	bsr := &common.BlockSubmissionResponse{RollupMetadata: rollupMetadata}
+
+	// in phase 1, only if the enclave is a sequencer, it can respond to shared secret requests
+	if e.isBackupSequencer(ctx) || e.isActiveSequencer(ctx) || e.sharedSecretService.IsGenesis() {
```

Comment:
I guess in HA scenarios this might lead to the host responding to the secret request twice (for each enclave) if we're not careful. I can make sure that's covered in a followup.